supreme/verifier/src/commonTest/kotlin/examples/Readme-Backend.kt

```diff
@@ -0,0 +1,79 @@
+package docs.service
+
+import at.asitplus.signum.indispensable.asn1.Asn1String
+import at.asitplus.signum.indispensable.asn1.Asn1Time
+import at.asitplus.signum.indispensable.pki.AttributeTypeAndValue
+import at.asitplus.signum.indispensable.pki.Pkcs10CertificationRequest
+import at.asitplus.signum.indispensable.pki.RelativeDistinguishedName
+import at.asitplus.signum.indispensable.pki.TbsCertificate
+import at.asitplus.signum.indispensable.pki.X509Certificate
+import at.asitplus.signum.indispensable.toX509SignatureAlgorithm
+import at.asitplus.signum.supreme.sign
+import at.asitplus.signum.supreme.sign.Signer
+import docs.config.minimal.verifier
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import io.ktor.server.plugins.contentnegotiation.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import kotlin.random.Random
+import kotlin.time.Clock
+import kotlin.time.Duration.Companion.days
+
+val PATH_CHALLENGE = "/api/v1/challenge"
+val PATH_ATTEST = "/api/v1/attest"
+
+val publicEndpoint: String = ""
+val signer =  Signer.Ephemeral {
+    ec { }
+}.getOrThrow()
+
+var issuerName = listOf(
+    RelativeDistinguishedName(
+        AttributeTypeAndValue.CommonName(
+            Asn1String.UTF8("Supreme Verifier")
+        )
+    )
+)
+var subjectName = listOf(
+    RelativeDistinguishedName(
+        AttributeTypeAndValue.CommonName(
+            Asn1String.UTF8("Supreme Client")
+        )
+    )
+)
+
+val caCert: X509Certificate = TODO()
+
+val server = embeddedServer(Netty, port = 8080) {
+   /*(1)!*/install(ContentNegotiation) { json() }
+
+  routing {
+ /*(2)!*/get(PATH_CHALLENGE) {
+      call.respond(
+      /*(3)!*/verifier.issueChallenge(/*(4)!*/"$publicEndpoint/$PATH_ATTEST")
+      )
+    }
+ /*(5)!*/post(PATH_ATTEST) {
+     /*(6)!*/val csr = Pkcs10CertificationRequest.decodeFromDer(call.receive<ByteArray>())
+        val result = verifier.verifyAttestation(csr) { csr ->
```

Comment:
do we really need to name the parameter of the lambda `csr`, since it shadows the outer name?

Reply:
good point!